### PR TITLE
[SIS-024] Home and Library header

### LINKED ISSUE #24: [SIS-024] Home and Library header
> As a user of the Sistema Lesson Planner, I want to be able to see the Sistema logo and access the settings page from the Home or Library page.

## Technical Information

_Note: When you start working on this, [[SIS-029] Add initial navigation and screens](https://github.com/uoftblueprint/sistema/pull/30) should be merged. Please ping @yuemily24 if it's not._ 

- [ ] Create a new directory `/src/components/` and a new file `Header.js`
- [ ] Implement a header component to be shared between the Library and Home page
- [ ] Implement a dummy "Settings" button. By dummy, we mean `onPress `doesn't have to do anything. (You may need to ask designers for the gear icon. Ignore the bear placeholder.)
- [ ] Place any logos, icons, and images in `/assets`
- [ ] Place the header in `Library.js` and `Home.js`

[Figma wireframe.](https://www.figma.com/file/kSZduD0A5hk8sHmvqiC7DE/Sistema-Hi-Fidelity-Designs-%2B-Design-System?node-id=97%3A13579&t=ourMQysTzVpxerGI-0) Here's the designer's design for the Home page (ignore the bear placeholder).

image

**Good Resources:**

[Image](https://reactnative.dev/docs/images)

- _Note:_ There are multiple ways of manipulating images in React. The default `Image` type may not be the best case for your needs. Play around with it and change types as needed.

[SVG docs](https://github.com/software-mansion/react-native-svg/blob/main/USAGE.md) -- for icons
1. Go to the Figma [hi-fidelity wireframe](https://www.figma.com/file/kSZduD0A5hk8sHmvqiC7DE/Sistema-Hi-Fidelity-Designs-%2B-Design-System?node-id=42%3A12578) to export the icons as SVGs. On the right sidebar, Design > Export > SVG. 
2. See example usage of SVGs [here](https://github.com/uoftblueprint/sistema/blob/097f08271e46bce430453430d04719ecff60a771/packages/mobile/assets/HomeIcon.js). 

[TouchableOpacity](https://reactnative.dev/docs/touchableopacity)
[Flexbox](https://reactnative.dev/docs/flexbox) for centering

### PR DESCRIPTION
**Pull request by MinGi-K**
# Description

Created custom header for library and home pages, and disabled the default header shown from the navigator. 

Ticket # (SIS-024)

## Type of change

Please delete options that are not relevant.

- [X ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ X ] My PR title is formatted as [SIS-\<issue number\>] \<issue name\> (eg. [SIS-010] Create UI button)
- [ X ] I have linked my issue/ticket on the project board to this PR (I think)?
- [ X ] I have sent my PR to #team-sistema-prs on Slack and requested two people to review
- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [X ] I have made corresponding changes to the documentation (if required)
- [ X] My changes generate no new warnings
- [ X] I have tested that my feature works as expected on Android and iOS. If you are a Windows dev, please request a Mac dev to run your branch. If it's a front-end change, I have attached screenshots of what the result looks like on both platforms.  (ONLY FOR ANDROID)
Screen Shot 2022-11-24 at 12 41 40 AM

### CODE REVIEW

**Review by Lemiex — APPROVED** [review]
Looks good Mingi! All files are in the directories they are suppose to be in and code is understandable.
A quick question: Do you have a screenshot for your Library header too? The screenshot attached is for Home if I'm not mistaken. Or did you not include it because they look the same.

**Comment by MinGi-K** [comment]
Yep, it is the same for both (empty without the header)!

**Comment by kurtislaw** [comment]
The branch also looks good to me. I agree with @Lemiex 's comments. Good job!

**Review by kurtislaw — APPROVED** [review]
The branch also looks good to me. I agree with @Lemiex 's comments. Good job!

**Comment by ramyzhang** [comment]
Awesome! We'll merge this in with the new repo setup once that's tested and merged. Thanks all 👯

**Comment by yuemily24** [comment]
@MinGi-K Were you able to test on iOS? You didn't need to since we're currently fixing the build, but I'm wondering because some of your file changes suggested that you somehow got iOS working. 

Can you revert/uncommit files that don't matter including:

- yarn.lock
- packages/mobile/ios/sistema.xcodeproj/project.pbxproj
- packages/mobile/ios/sistema.xcworkspace/contents.xcworkspacedata
- packages/mobile/ios/sistema.xcodeproj/project.pbxproj
- package-lock.json
- packages/mobile/ios/Podfile.lock
- Any `.idea/` files

File: packages/mobile/assets/OptionIcon.js
Comment (yuemily24) [inline packages/mobile/assets/OptionIcon.js:2]: Can you get rid of these blank lines? Running `yarn run lint` should also fix any trivial formatting.

File: packages/mobile/assets/SistemaLogo.js
Comment (yuemily24) [inline packages/mobile/assets/SistemaLogo.js:6]: Same thing here. `yarn run lint`

File: packages/mobile/src/components/Header.js
Comment (yuemily24) [inline packages/mobile/src/components/Header.js:60]: Is there a slightly easier way to do this? Like adding `marginBottom`. If it works though, it works!

File: packages/mobile/src/home/Home.js
Comment (yuemily24) [inline packages/mobile/src/home/Home.js:7]: Can you revert this back to `SafeAreaView`? We'll be using it as a parent container for all our pages for this reason:

> The purpose of SafeAreaView is to render content within the safe area boundaries of a device. It is currently only applicable to iOS devices with iOS version 11 or later.
> 
> SafeAreaView renders nested content and automatically applies padding to reflect the portion of the view that is not covered by navigation bars, tab bars, toolbars, and other ancestor views. Moreover, and most importantly, Safe Area's paddings reflect the physical limitation of the screen, such as rounded corners or camera notches (i.e. the sensor housing area on iPhone 13).

Read more about SafeAreaView [here](https://reactnative.dev/docs/safeareaview).

File: packages/mobile/src/home/Home.js
Comment (yuemily24) [inline packages/mobile/src/home/Home.js:14]: Confused why there's empty `View` container here. The header component doesn't need to be sticky if that's what you're worried about.

File: packages/mobile/src/library/Library.js
Comment (yuemily24) [inline packages/mobile/src/library/Library.js:7]: Once again, please revert to `SafeAreaView` for above reasons.

File: packages/mobile/src/library/Library.js
Comment (yuemily24) [inline packages/mobile/src/library/Library.js:14]: Same comment confused why this is here

File: packages/mobile/src/components/Header.js
Comment (yuemily24) [inline packages/mobile/src/components/Header.js:37]: Instead of doing this, you might be able to use [alignSelf](https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-align-self) for both the settings icon (shift it to the end) and logo (align it center). It'll probably be easier and more fluid than inserting a blank view to force spacing (try to avoid if you can).

**Review by yuemily24 — CHANGES_REQUESTED** [review]
I basically have duplicate comments for `Home` and `Library` screens, please address! Also, this [flexbox cheatsheet](https://css-tricks.com/snippets/css/a-guide-to-flexbox/#aa-align-self) is really helpful for me. It's meant for web dev, but you can basically copy properties by converting names to camel case. Otherwise, code is nicely commented, good job!

Also, _when you commit from now on, please run `yarn run cm` in the root folder_. It'll run a commit script for you to format your commit messages nicely.

**Pull request closed without merging**
